Mathlib/Algebra/Star/Order.lean

```diff
@@ -147,6 +147,11 @@ theorem nonneg_iff [NonUnitalSemiring R] [PartialOrder R] [StarOrderedRing R] {x
   simp only [le_iff, zero_add, exists_eq_right']
 #align star_ordered_ring.nonneg_iff StarOrderedRing.nonneg_iff
 
+theorem le_iff' [NonUnitalSemiring R] [PartialOrder R] [StarOrderedRing R] {x y : R} :
```

Comment:
Isn't this `exists_add_iff_le` or something? Maybe I've imagined that we actually have that

Reply:
We have [le_iff_exists_add](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Monoid/Canonical/Defs.html#le_iff_exists_add), but this has no nonnegativity condition and requires a `CanonicallyOrderedAddMonoid`, and we have [le_iff_exists_nonneg_add](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Ring/Defs.html#le_iff_exists_nonneg_add), but that requries an `OrderedRing`.

However, it should be possible to add a non-strict version of the lemma [exists_pos_add_of_lt'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Monoid/Canonical/Defs.html#exists_pos_add_of_lt') under a `ContravariantClass α α (· + ·) (· ≤ ·)`. That being said, it doesn't really help us here because I think we can't conclude that without the `OrderedAddCommGroup` instance when we have a `NonUnitalRing` (instead of a `NonUnitalSemiring`).

One possible solution would be to hardcode the `ContravariantClass` condition into the definition of `StarOrderedRing` (I can't think of any counterexamples where we would want a `StarOrderedRing` class but it doesn't satisfy the `ContravariantClass` condition, but perhaps I'm just not thinking hard enough), but that requires another change to the definition.